src/jit/ee_il_dll.cpp

```diff
@@ -129,15 +112,13 @@ BOOL WINAPI     DllMain(HANDLE hInstance, DWORD dwReason, LPVOID pvReserved)
     {
         g_hInst = (HINSTANCE)hInstance;
         DisableThreadLibraryCalls((HINSTANCE)hInstance);
-#ifdef SELF_NO_HOST
-        jitOnDllProcessAttach();
-        g_fClrCallbacksInit = TRUE;
+#if defined(SELF_NO_HOST) && COR_JIT_EE_VERSION <= 460
+		jitStartup(JitHost::getJitHost());
 #endif
     }
-    else if (dwReason == DLL_PROCESS_DETACH)
+    else if (dwReason == DLL_PROCESS_DETACH && g_jitInitialized)
```

Comment:
Yes: the JIT is not automatically initialized by DllMain. It will not have been initialized if it was loaded and unloaded without an intervening call to jitStartup.